core/src/main/java/org/eclipse/hono/util/TenantObject.java

```diff
@@ -317,13 +359,20 @@ public JsonObject getAdapterConfiguration(final String type) {
      *                              copied into a new list in order to prevent modification
      *                              of the list after this method has been invoked.
      * @return This tenant for command chaining.
+     * @throws IllegalArgumentException if the given configurations does not contain a <em>type</em> or
+     *                                  if any of the already existing adapters has the same <em>type</em> or
+     *                                  if error parsing the given configuration.
+     * @deprecated Use {@link #setAdapters(List adapters)} instead.
      */
+    @Deprecated
     @JsonProperty(TenantConstants.FIELD_ADAPTERS)
     public TenantObject setAdapterConfigurations(final List<Map<String, Object>> configurations) {
         if (configurations == null) {
-            this.adapterConfigurations = null;
+            this.adapters = null;
         } else {
-            this.adapterConfigurations = new LinkedList<>(configurations);
+            configurations.stream()
+                    .map(JsonObject::mapFrom)
+                    .forEach(this::addAdapterConfiguration);
```

Comment:
this does not replace the existing adapter configs with the given ones, does it?

Reply:
Thanks for pointing out. This has been fixed now.